javaagent-exporters/prometheus/src/main/java/io/opentelemetry/javaagent/exporters/prometheus/PrometheusMetricCollector.java

```diff
@@ -39,22 +37,18 @@
   // The empty address equals to any address
   private static final String DEFAULT_HOST = "0.0.0.0";
 
-  private final MeterSdkProvider meterSdkProvider = OpenTelemetrySdk.getMeterProvider();
-
   @Override
-  public MetricExporter fromConfig(ExporterConfig config) {
-    PrometheusCollector.newBuilder()
-        .setMetricProducer(meterSdkProvider.getMetricProducer())
-        .buildAndRegister();
+  public void init(MetricProducer producer, ExporterConfig config) {
+
+    PrometheusCollector.newBuilder().setMetricProducer(producer).buildAndRegister();
     try {
       int port = config.getInt(PORT_CONF_PROP_NAME, DEFAULT_PORT);
       String host = config.getString(HOST_CONF_PROP_NAME, DEFAULT_HOST);
       log.info("Creating Prometheus exporter on host: '{}' and port: '{}'", host, port);
-      HTTPServer server = new HTTPServer(host, port);
-      return new PrometheusExporter(server);
+      // TODO close the server
```

Comment:
nice thing about javaagent is we don't have to close, though we do have to worry about creating non-daemon threads, can you check that? e.g. use the javaagent w/ prometheus exporter from a command-line app and make sure we don't prevent the app from exiting and the end of `main`
```suggestion
```

Reply:
Good catch, there is a `daemon` param on the constructor.